tests.py

```diff
@@ -0,0 +1,105 @@
+import unittest
+from zoo_project import Cage, Animal, Zoo
+
+
+class TestZooCreation(unittest.TestCase):
+    """Just creates a Zoo and checks the name is the same one we used at creation time.
+    """
+
+    def test_zoo(self):
+        zoo = Zoo("London Zoo")
+        self.assertEqual(zoo.name, 'London Zoo')
+
+
+class TestCageCreation(unittest.TestCase):
+    """Creates a cage
+    """
+
+    def test_cage(self):
+        cage = Cage()
+
+
+class TestAnimalCreation(unittest.TestCase):
+    """Creates an animal and checks that the name corresponds to the one we used at creation time.
+    """
+
+    def test_animal(self):
+        freddie_the_lion = Animal("Freddie", "Lion")
+        self.assertEqual(freddie_the_lion.animal_name, 'Freddie')
+
+
+class TestAddAnimalToCage(unittest.TestCase):
+    """Creates an animal, adds it to the cage and checks that the cage contains an animal with the right name in the text.
+    """
+
+    def test_add_animal_to_cage(self):
+        freddie_the_lion = Animal("Freddie", "Lion")
+        cage = Cage()
+        cage.add_animal(freddie_the_lion)
+        self.assertEqual(str(cage.animals[0]), 'Freddie the Lion')
+
+
+class TestDeleteAnimalFromCage(unittest.TestCase):
+    """Creates and adds two animals to a cage, then deletes one and expects the number of animals in the cage to be 1.
+    """
+
+    def test_delete_animal_from_cage(self):
+        freddie_the_lion = Animal("Freddie", "Lion")
+        turty_the_turtle = Animal("Turty", "Turtle")
+        cage = Cage()
+        cage.add_animal(freddie_the_lion)
+        cage.add_animal(turty_the_turtle)
+        cage.delete_animal(freddie_the_lion)
```

Comment:
There may be better names than `delete_animal`.  Conceptually I would expect that to remove the animal from existence, not just from the cage.  Maybe `remove_animal` would be better?